### PR TITLE
Update types--validation@UNlvRp6k3_RDoTAAIEfJ1.md

### PR DESCRIPTION
**Pull request by Primis1**
Add description for Next.JS Types & Validation node

comment:
Guys you did help me much in the past, so I'm grateful and think u're cool

### CODE REVIEW

**Review by dansholds — CHANGES_REQUESTED** [review]
Thanks for the kind words!

Let's keep this content to one concise paragraph and don't forget to add

`Learn more from the following resources`

Before the content links.

Remember to check the contribution docs and other topics in the project.

**Comment by Primis1** [comment]
Thank you for a review, did the article formatting and tiny fixes in spelling, hope it is going to work now)

**Comment by Primis1** [comment]
Thank you for a review, did the article formatting and tiny fixes in spelling, hope it is going to work now)

File: src/data/roadmaps/react/content/types--validation@UNlvRp6k3_RDoTAAIEfJ1.md [hunk 1]
```diff
@@ -1 +1,11 @@
...
+
+In React.js, there are two major/optimal tools for types & validation: 
+First is **Typescript**. It allows small/medium/large organization & team don't shot itself in a foot during developing. It provides a static type system like in most low-level languages on a **developing** side. 
+**_Mention that TypeScript, basically disappears after interpretation process. So we can't rely on it to validate a client's input._**
+Second one **Zod**. On the other hand, is fresh and powerful **_TypeScript only_**(supports TS instead of JS) validation library. It allows us to cover weak points of typescript. It can exist wherever **TypeScript can't**. Zod can validate: form input, local storage, API contracts and much more 
+
+Visit the following resources to learn more:
+
+- [@article@TypeScript Essentials](https://www.freecodecamp.org/news/why-use-static-types-in-javascript-part-1-8382da1e0adb/)
+- [@article@When Use Zod](https://www.totaltypescript.com/when-should-you-use-zod)
```
Comment (kamranahmedse) [inline src/data/roadmaps/react/content/types--validation@UNlvRp6k3_RDoTAAIEfJ1.md:11]:
```suggestion
# Types & Validation

**Typescript** provides a static type system that helps you in avoiding mistakes during the development and provides other features (e.g. IDE support) that help you improve your productivity. 

One thing you should note is that TypeScript can only help you avoid mistakes during the development. We can't rely on it to validate a client's input. **Zod** is a powerful validation library that allows us to validate: form input, local storage, API contracts and much more using their typesafe implementation.

Visit the following resources to learn more:

- [@article@What is TypeScript and why would I use it in place of JavaScript?](https://stackoverflow.com/questions/12694530/what-is-typescript-and-why-would-i-use-it-in-place-of-javascript)
- [@article@When Use Zod](https://www.totaltypescript.com/when-should-you-use-zod)
```

**Pull request merged**